iot/http_example/http_client_example.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2017, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** NodeJS sample of connecting to Google Cloud IoT Core via HTTP. */
+
+'use strict';
+
+const fs = require('fs');
+const google = require('googleapis');
+const jwt = require('jsonwebtoken');
+
+// Create a Cloud IoT Core JWT for the given project id, signed with the given
+// private key.
+function createJwt (projectId, privateKeyFile, algorithm) {
+  // Create a JWT to authenticate this device. The device will be disconnected
+  // after the token expires, and will have to reconnect with a new token. The
+  // audience field should always be set to the GCP project id.
+  const token = {
+    'iat': parseInt(Date.now() / 1000),
+    'exp': parseInt(Date.now() / 1000) + 20 * 60,  // 20 minutes
+    'aud': projectId
+  };
+  const privateKey = fs.readFileSync(privateKeyFile);
+  return jwt.sign(token, privateKey, { algorithm: algorithm });
+}
+
+// Publish numMessages messages asynchronously, starting from message
+// messageCount.
+function publishAsync (bearer, client, messageCount, message, projectId, cloudRegion, registryId, deviceId) {
```

Comment:
The order of arguments is a bit inconsistent:

for this function: `(bearer, client, messageCount, message, projectId, cloudRegion, registryId, deviceId)`

for `getDeviceState` in `manager.js` it is `(client, deviceId, registryId, projectId, cloudRegion)`